templates/etc/haproxy/frontend.cfg.j2

```diff
@@ -30,6 +30,14 @@ frontend {{ frontend.name }}
 {% endfor %}
 {% endif %}
 
+{% if frontend.acls is defined %}
+{% for acl in frontend.acls %}
+  acl host_{{acl.name}} hdr(host) -i {{acl.host}}
+  use_backend {{acl.backend}} if host_{{acl.name}}
```

Comment:
I think this is too detailed we will never be able to implement all possible options here. I think that it much better to do this simple and ask user to just write whole ACL lines.